drivers/clock_control/nrf5_power_clock.c

```diff
@@ -253,7 +260,12 @@ static void _power_clock_isr(void *arg)
 	done = (NRF_CLOCK->EVENTS_DONE != 0);
 	ctto = (NRF_CLOCK->EVENTS_CTTO != 0);
 
-	__ASSERT_NO_MSG(pof || hf || hf_intenset || lf || done || ctto);
+	usb_detected = (NRF_POWER->EVENTS_USBDETECTED != 0);
```

Comment:
conditional compile under CONFIG_USB.

Reply:
Yes, I had forgotten this one. Will do. Although CONFIG_USB should do the job, shouldn't it be CONFIG_USB_NRF52 rather?

Reply:
Maybe as below (tho for now only nRF52840 SoC will be supported)

```
#if defined(CONFIG_USB)
#if defined(CONFIG_SOC_NRF52840)
....
#endif
#endif
```